test/tests/functional/pbs_job_array.py

```diff
@@ -165,12 +165,26 @@ def test_running_subjob_survive_restart(self):
         self.server.expect(JOB, {'job_state': 'X'},
                            j.create_subjob_id(j_id, 1), max_attempts=1)
 
-        # 6. subjob 2 should be R
-        self.server.expect(JOB, {'job_state': 'R'}, subjid_2, max_attempts=1)
+        subjob2_finished = False
+        # 6. subjob 2 should be in R or X state
+        try:
+            self.server.expect(JOB, {'job_state': 'R'}, subjid_2,
+                               max_attempts=1)
+        except PtlExpectError:
+            self.server.expect(JOB, {'job_state': 'X'}, subjid_2,
+                               max_attempts=1)
```

Comment:
Since you do max_attempts=1, it might be better to call status(), and check the job_state.  The purpose of expect() is to check over time.  You only check once, so it isn't any different than a status(), except the two expect() calls will call status() twice.